client/lib/eventHubSender.ts

```diff
@@ -248,6 +248,10 @@ export class EventHubSender extends LinkEntity {
         throw new Error("data is required and it must be of type object.");
       }
 
+      if (data.partitionKey){
+        data.partitionKey = (typeof data.partitionKey === "number") ? (+data.partitionKey).toString() : data.partitionKey;
```

Comment:
I wonder if we should restrict this to just number.

What exactly happens when we pass a non string value as partition key?
Is it our sdk that fails to pass it along?
Or is it Event Hubs itself that ignores the value?

Reply:
Even without this change:
``` 
if (data.partitionKey){
        data.partitionKey = (typeof data.partitionKey === "number") ? (+data.partitionKey).toString() : data.partitionKey;
      }
```
It works fine If we pass data like this:
```
 const data = { body: { "message": "Hello World partitioned key as a number" }, partitionKey: 123};
```
It means that Event hub ignores the value.

Reply:
we can change above code like this:
```
 if (data.partitionKey){
        data.partitionKey = (typeof data.partitionKey !== "string") ? (+data.partitionKey).toString() : data.partitionKey;
      }
```

Reply:
> It works fine If we pass data like this:

I was not expecting Event Hubs to throw any error.

My question was when you passed say 123 as partition key, does Event Hubs use that as partition key?

Reply:
If we don't pass partition Id while sending, partition key say 123 guarantees that messages end up in a existing partition on the event hub.